src/Admin/AdminHelper.php

```diff
@@ -39,13 +40,37 @@ class AdminHelper
     private const FORM_FIELD_DELETE = '_delete';
 
     /**
+     * NEXT_MAJOR: Remove this property.
+     *
      * @var Pool
      */
     protected $pool;
 
-    public function __construct(Pool $pool)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    /**
+     * NEXT_MAJOR: Make $propertyAccessor mandatory and remove $pool.
+     */
+    public function __construct(Pool $pool, ?PropertyAccessorInterface $propertyAccessor = null)
```

Comment:
Pool won't be needed as a dependency in the next major so we should allow to pass `propertyAccessor` as the first argument and Pool as the second one no ?

Reply:
I've change the order also for SonataAdminExtension in another commit, I'll squash when this is right.